cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/TimeScheduler.java

```diff
@@ -147,35 +150,45 @@ public void schedule(Id.Program programId, SchedulableProgramType programType, I
       for (Trigger trigger : scheduler.getTriggersOfJob(getJobKey(program, programType))) {
         scheduleIds.add(trigger.getKey().getName());
       }
-    }   catch (SchedulerException e) {
-      throw Throwables.propagate(e);
+    }   catch (org.quartz.SchedulerException e) {
+      throw new SchedulerException(e);
     }
     return scheduleIds;
   }
 
 
   @Override
-  public void suspendSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName) {
+  public void suspendSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName)
+    throws NotFoundException, SchedulerException {
     checkInitialized();
     try {
       scheduler.pauseTrigger(new TriggerKey(getScheduleId(program, programType, scheduleName)));
-    } catch (SchedulerException e) {
-      throw Throwables.propagate(e);
+    } catch (org.quartz.SchedulerException e) {
+      throw new SchedulerException(e);
     }
   }
 
   @Override
-  public void resumeSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName) {
+  public void resumeSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName)
+    throws NotFoundException, SchedulerException {
     checkInitialized();
     try {
       scheduler.resumeTrigger(new TriggerKey(getScheduleId(program, programType, scheduleName)));
-    } catch (SchedulerException e) {
-      throw Throwables.propagate(e);
+    } catch (org.quartz.SchedulerException e) {
+      throw new SchedulerException(e);
     }
   }
 
   @Override
-  public void deleteSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName) {
+  public void updateSchedule(Id.Program program, SchedulableProgramType programType, Schedule schedule)
```

Comment:
you will change this to update the schedule without deleting, in the next PR(s)? To make sure we do not miss a scheduled run because deleting a schedule also deletes its last run statistics.